### PR TITLE
Change staking tooltip to display frequency

### PR DESCRIPTION
**Pull request by jamescowens**
This addresses #1442.

### CODE REVIEW

**Comment by jamescowens** [comment]
Screenshot...
image

File: src/qt/bitcoingui.cpp [hunk 1]
```diff
@@ -1425,25 +1425,40 @@ void BitcoinGUI::timerfire()
...
-        else
+
+    if (frequency >= next_unit_threshold * 24.0)
     {
-            text = tr("%n day(s)", "", nEstimateTime/(60*60*24));
+        frequency /= 24.0;
+        unit = "hour";
     }
+
+    text = tr("%1 times per %2").arg(QString(RoundToString(frequency, 2).c_str())).arg(unit);
```
Comment (cyrossignol) [inline src/qt/bitcoingui.cpp:1462]: We'll need to wrap `unit` in `tr()` as well localize it. The placeholder substitutions (`%2`) are not automatically translated.

File: src/qt/bitcoingui.cpp [hunk 2]
```diff
@@ -1425,25 +1425,40 @@ void BitcoinGUI::timerfire()
...
-QString BitcoinGUI::GetEstimatedTime(unsigned int nEstimateTime)
+QString BitcoinGUI::GetEstimatedStakingFrequency(unsigned int nEstimateTime)
 {
     QString text;
-    if (nEstimateTime < 60)
+
+    if (!nEstimateTime)
     {
-            text = tr("%n second(s)", "", nEstimateTime);
+        text = tr("undefined");
```
Comment (cyrossignol) [inline src/qt/bitcoingui.cpp:1434]: I think you intended to put a `return` after this branch.

File: src/main.h [hunk 3]
```diff
@@ -260,10 +260,10 @@ bool IsSuperBlock(CBlockIndex* pIndex);
 
 double GetEstimatedNetworkWeight(unsigned int nPoSInterval = 40);
 double GetAverageDifficulty(unsigned int nPoSInterval = 40);
-//double GetEstimatedTimetoStake(unsigned int nPoSInterval = 40, double dConfidence = 0.8);
 // Note that dDiff cannot be = 0 normally. This is set as default because you can't specify the output of
 // GetAverageDifficulty(nPosInterval) = to dDiff here.
-double GetEstimatedTimetoStake(double dDiff = 0.0, double dConfidence = 0.8);
+// The defeult confidence is 1-1/e which is the mean for the geometric distribution.
+double GetEstimatedTimetoStake(double dDiff = 0.0, double dConfidence = 0.632120558829);
```
Comment (cyrossignol) [inline src/main.h]: Can we define the new default value for `dConfidence` in a constant to formalize it?

File: src/qt/bitcoingui.cpp (hunk 1 above)
Comment (jamescowens) [reply src/qt/bitcoingui.cpp:1462]: Ah yes. thanks for reminding me.

File: src/qt/bitcoingui.cpp (hunk 2 above)
Comment (jamescowens) [reply src/qt/bitcoingui.cpp:1434]: yep

File: src/main.h (hunk 3 above)
Comment (jamescowens) [reply src/main.h]: Sure.

**Review by cyrossignol — APPROVED** [review]

**Pull request merged**